packages/libs/hd-wallet/src/index.ts

```diff
@@ -1 +1,253 @@
+  } else {
+    seed = encryptOrEncodeSeed(seedBuffer);
+  }
+  return {
+    seedBuffer,
+    seed,
+  };
+}
+
+/**
+ * Generates a mnemonic phrase using the BIP39 protocol with a specified wordlist.
+ *
+ * @returns {string} A valid BIP39 mnemonic phrase.
+ * @throws {Error} If the generated mnemonic is invalid.
+ */
+export function kadenaGenMnemonic(): string {
+  return bip39.generateMnemonic(wordlist);
+}
+
+/**
+ * Generates a key pair from a seed buffer and an index or range of indices, and optionally encrypts the private key.
+ *
+ * @param {Uint8Array} seedBuffer - The seed buffer to use for key generation.
+ * @param {number | [number, number]} indexOrRange - Either a single index or a tuple with start and end indices for key pair generation.
+ * @param {string} [password] - Optional password for encrypting the private key.
+ * @returns {([string, string] | [string, string][])} - Depending on the input, either a tuple for a single key pair or an array of tuples for a range of key pairs, with the private key encrypted if a password is provided.
+ * @throws {Error} Throws an error if the seed buffer is not provided, if the indices are invalid, or if encryption fails.
+ */
+export function kadenaGenKeypair(
+  seedBuffer: Uint8Array,
+  indexOrRange: number | [number, number],
+  password?: string,
+): [string, string] | [string, string][] {
+  if (seedBuffer === undefined) throw new Error('No seed provided.');
+
+  if (Array.isArray(indexOrRange)) {
+    const [start, end] = indexOrRange;
+    if (start > end)
+      throw new Error(
+        'Invalid range: start index must be less than or equal to end index.',
+      );
+
+    const keyPairs: [string, string][] = [];
+    for (let i = start; i <= end; i++) {
+      const { publicKey, privateKey } = deriveKeyPair(seedBuffer, i);
+      const privatekeyToUse =
+        password !== undefined
+          ? encryptPrivateKey(Buffer.from(privateKey, 'hex'), password)
+          : privateKey;
+      keyPairs.push([publicKey, privatekeyToUse]);
+    }
+    return keyPairs;
+  } else {
+    const { publicKey, privateKey } = deriveKeyPair(seedBuffer, indexOrRange);
+
+    const privatekeyToUse =
+      password !== undefined
+        ? encryptPrivateKey(Buffer.from(privateKey, 'hex'), password)
+        : privateKey;
+    return [publicKey, privatekeyToUse];
+  }
+}
+
+/**
+ * Decrypts an encrypted private key using the provided password.
+ * This function is a wrapper for the internal decryption logic, intended
+ * for public-facing API usage where the private key encryption follows
+ *
+ * @param {string} encryptedPrivateKey - The encrypted private key as a Base64 encoded string.
+ * @param {string} password - The password used to encrypt the private key.
+ * @returns {Uint8Array} The decrypted private key.
+ * @throws {Error} Throws an error if decryption fails.
+ */
+export function kadenaDecryptPrivateKey(
+  encryptedPrivateKey: string,
+  password: string,
+): Uint8Array {
```

Comment:
Create a generic encrypt and decrypt function that both use random salts and include them in the encrypted output
```
export function kadenaDecrypt(
  password: string,
  encryptedMessage: string,
): Uint8Array
```

```
export function kadenaEncrypt(
  password: string,
  message: string,
): Uint8Array
```